feat(apple): Add docs for Swift Error description (#6805)
getsentry/sentry-docs: src/platform-includes/capture-error/apple.mdx

[SentrySDK captureException:exception];
```

### Swift Errors

For Swift Errors conforming to the [Error Protocol](https://developer.apple.com/documentation/swift/error) the SDK
sends the domain, code and the description of the Swift error.

> Before [sentry-cocoa 8.7.0](https://github.com/getsentry/sentry-cocoa/blob/main/CHANGELOG.md#870) the SDK sends only

Reviewer: ```suggestion
For older versions of the SDK, prior to [sentry-cocoa 8.7.0](https://github.com/getsentry/sentry-cocoa/blob/main/CHANGELOG.md#870) the SDK only sends
```

> the domain and error code.

Reviewer: Is there a reason for these funny line breaks? It doesn't affect the output, but it's a little annoying

Author: No, not a reason. I removed them.

```Swift
enum LoginError: Error {
    case wrongUser(id: String)
    case wrongPassword
}

SentrySDK.capture(error: LoginError.wrongUser("12345678"))
```

> For the Swift error above Sentry displays:

Reviewer: ```suggestion
For the Swift error above Sentry would display the following, based on SDK version:
```

| sentry-cocoa SDK | Title | Description |
| ----------- | ----------- | ----------- |
| Since 8.7.0 | `LoginError` | `wrongUser(id: "12345678") (Code: 1)` |
| Before 8.7.0 | `LoginError` | `Code: 1` |

### Customizing Error Descriptions

This feature is available on [sentry-cocoa 7.25.0](https://github.com/getsentry/sentry-cocoa/blob/master/CHANGELOG.md#7250) and above.

> You may want to provide a custom description to make identifying the error in the _Issues_ page easier. For `NSError` values, this can be done by adding a description to the `userInfo` dictionary with the key `NSDebugDescriptionErrorKey`.

Reviewer: ```suggestion
You may want to provide a custom description to make identifying the error in the **Issues** page easier. For `NSError` values, you can do this by adding a description to the `userInfo` dictionary with the key `NSDebugDescriptionErrorKey`.
```

bold page titles according to style guide

    }
}
```

<PlatformSection supported={["apple.macos"]}>

## Capturing Uncaught Exceptions in macOS

By default, macOS applications do not crash whenever an uncaught exception occurs. To enable this with Sentry:

1. Open the application's `Info.plist` file
2. Search for `Principal class` (the entry is expected to be `NSApplication`)
3. Replace `NSApplication` with `SentryCrashExceptionApplication`

> </PlatformSection>

Author: I moved this existing section down here.
